provider/kv_test.go

```diff
@@ -410,6 +410,14 @@ func TestKVLoadConfig(t *testing.T) {
 					Key:   "traefik/backends/backend.with.dot.too/servers/server.with.dot/weight",
 					Value: []byte("0"),
 				},
+				{
+					Key:   "traefik/backends/backend.with.dot.too/servers/server.with.dot.without.url",
+					Value: []byte(""),
+				},
+				{
+					Key:   "traefik/backends/backend.with.dot.too/servers/server.with.dot.without.url/weight",
+					Value: []byte("0"),
+				},
```

Comment:
Hmm, I'm not sure how it can/should be implemented. We can't rely on the `Mock` `Error` bool, as that cause `List` to error out.

I think we need to crate our own `KVPair` type with a `error` "field", but that seems kind of messy. Do you have a simpler idea?